.gitignore

```diff
@@ -38,3 +38,16 @@ Goopfile.lock
 venv/
 rultor_secrets.sh
 .vscode
+
+#Ignore tags created by ctags, etags and gtags
+TAGS
+.TAGS
+!TAGS/
+tags
+.tags
+!tags/
+gtags.files
+GTAGS
+GRTAGS
+GPATH
+GSYMS
```

Comment:
Can't we have a single pattern to ignore all the temporary files created by vim instead of listing them one by one?
see https://stackoverflow.com/questions/4824188/git-ignore-vim-temporary-files

Reply:
Yes you can but not in this case. I don't think so. Unlike the `.swp` files and the like, tag files are not temporary files in vim. They are real editable permanent files that contain tags to certain function and variable definitions present in the project which are then used during navigation.

Reply:
But those are quite many... are these really all temporary files by vim? I've seen people provide only `tags` in some gitignore-files.

Reply:
Yes only tags might work too. It will for my case. I just included others in case someone else is using tags configured in a different way. I searched a little bit and took help of [this suggested .gitignore](https://github.com/github/gitignore/blob/master/Global/Tags.gitignore) file.

Reply:
@Makman2 you think I should change it?

Reply:
@nityeshaga , maybe. Take a look at this Vim.gitignore file where they have only used `tags` for auto generated tag files https://github.com/github/gitignore/blob/master/Global/Vim.gitignore

Reply:
If someone is using the `gtags` program for creating such tags, he/she may have these 3 files- 
* `GPATH`, 
* `GRTAGS` and 
* `GTAGS`

[Further reading](http://www.gnu.org/software/global/globaldoc_toc.html)

Similarly other files may also be generated based on some specific settings that may be used by a user. Actually there are 3 programs that are generally used for creating such tags files- `ctags` (probably the most popular), `etags` and `gtags`. AFAIK `ctags`, in its most basic setting, will generate a file named `tags`.

So, the question comes down to whether we want our gitignore to work with all these different settings or just the most common one?

Reply:
Ah okay those are different tagging systems, sounds okay :+1: